src/core/include/mp-units/numbers.h

```diff
@@ -0,0 +1,294 @@
+template<typename T, typename U>
+concept compound_point_space_for = point_space_for<T, U> && detail::compound_subtraction_with<T, U>;
+template<typename T>
+concept point_space = compound_point_space_for<T, number_difference_t<T>>;
+
+template<typename T, typename U>
+concept vector_space_for = point_space_for<U, T>;
+template<typename T, typename U>
+concept compound_vector_space_for = compound_point_space_for<U, T>;
+
+namespace detail {
+
+template<typename T>
+concept weak_scalar = std::common_with<T, number_difference_t<T>> && point_space<T> && negative<T>;
+
+template<typename T, typename U>
+concept scales_with =
+  std::common_with<U, number_scalar_t<T>> && weak_scalar<U> && multiplication_with<T, U, T> && set_with_inverse<U>;
+template<typename T, typename U>
+concept compound_scales_with = scales_with<T, U> && detail::compound_multiplication_with<T, U>;
+
+}  // namespace detail
+
+template<typename T, typename U>
+concept scalar_for = detail::scales_with<U, T>;
+template<typename T, typename U>
+concept field_for = scalar_for<T, U> && detail::division_with<U, T>;
+template<typename T, typename U>
+concept compound_scalar_for = detail::compound_scales_with<U, T>;
+template<typename T, typename U>
+concept compound_field_for = compound_scalar_for<T, U> && detail::compound_division_with<U, T>;
+
+template<typename T>
+concept vector_space = point_space<T> && detail::compound_scales_with<T, number_scalar_t<T>>;
+template<typename T>
+concept f_vector_space = vector_space<T> && detail::compound_division_with<T, number_scalar_t<T>>;
+
+template<typename T>
+concept field_number = f_vector_space<T> && detail::compound_scales_with<T, T>;
+
+template<typename T>
+concept field_number_line = field_number<T> && number_line<T>;
+
+template<typename T>
+concept scalar_number = field_number<T> && (field_number_line<T> || is_complex_number_v<T>);
+
+template<typename T>
+struct is_number<const T> : is_number<T> {};
+template<typename T, typename U>
+struct is_number<std::chrono::time_point<T, U>> : std::true_type {};
+#if __cpp_lib_chrono >= 201803
+template<>
+struct is_number<std::chrono::day> : std::true_type {};
+template<>
+struct is_number<std::chrono::month> : std::true_type {};
+template<>
+struct is_number<std::chrono::year> : std::true_type {};
+template<>
+struct is_number<std::chrono::weekday> : std::true_type {};
+template<>
+struct is_number<std::chrono::year_month> : std::true_type {};
+#endif
+
+template<typename T>
+struct is_complex_number<const T> : is_complex_number<T> {};
+template<typename T>
+struct is_complex_number<std::complex<T>> : std::true_type {};
+
+namespace detail {
+
+template<typename T>
+concept inferable_identities = std::common_with<T, number_difference_t<T>> && std::constructible_from<T, int>;
+
+template<detail::inferable_identities T>
+struct inferred_number_zero<T> {
+  auto static constexpr value = T(0);
```

Comment:
```suggestion
  static constexpr auto value = T(0);
```

It is consistent with the rest of the library and probably looks better in case it is constrained with a concept.